File: frontend/apps/web-antd/src/api/admin/cache.ts
/**
 * Cache management API / 缓存管理 API
 * Backend: /admin/cache/*
 */
import { requestClient } from '#/utils/request';

const API_PREFIX = '/admin/cache';

// ── Types / 类型定义 ──

/** Cache category summary / 缓存分类摘要 */
export interface CacheCategorySummary {
  category: string;
  label: string;
  key_count: number;
  size_bytes: number;
  size_human: string;
}

/** Cache summary response / 缓存统计响应 */
export interface CacheSummaryResponse {
  categories: CacheCategorySummary[];
  total_size_bytes: number;
  total_size_human: string;
}

/** Cache clear request / 缓存清理请求 */
export interface CacheClearRequest {
  categories: string[];
}

/** Cache clear response / 缓存清理响应 */
export interface CacheClearResponse {
  cleared_categories: string[];
  cleared_keys: number;
  cleared_size_bytes: number;
  cleared_size_human: string;
  duration_ms: number;
}

// ── API Functions / API 接口 ──

/** Get cache summary / 获取缓存统计摘要 */
export function getCacheSummaryApi() {
  return requestClient.get<CacheSummaryResponse>(`${API_PREFIX}/summary`);
}

/** Clear specified category cache / 清理指定分类缓存 */
export function clearCacheApi(data: CacheClearRequest) {
  return requestClient.post<CacheClearResponse>(`${API_PREFIX}/clear`, data);
}

File: backend/app/schemas/system/cache.py
"""
缓存管理相关 Schema
"""

from pydantic import Field, field_validator

from app.core.base_schema import BaseSchema
from app.enums.cache import CacheCategoryEnum


class CacheCategorySummary(BaseSchema):
    """单个缓存分类的统计信息"""

    category: str = Field(..., description="Cache category code")
    label: str = Field(..., description="Cache category i18n label")
    key_count: int = Field(0, description="Number of keys or files")
    size_bytes: int = Field(0, description="Total size in bytes")
    size_human: str = Field("0 B", description="Human-readable size")


class CacheSummaryResponse(BaseSchema):
    """缓存统计汇总响应"""

    categories: list[CacheCategorySummary] = Field(
        default_factory=list, description="Cache category summaries"
    )
    total_size_bytes: int = Field(0, description="Total size in bytes")
    total_size_human: str = Field("0 B", description="Human-readable total size")


class CacheClearRequest(BaseSchema):
    """缓存清理请求"""

    categories: list[str] = Field(
        ..., min_length=1, description="List of cache category codes to clear"
    )

    @field_validator("categories", mode="before")
    @classmethod
    def validate_categories(cls, v: list[str]) -> list[str]:
        """Validate that all categories are valid CacheCategoryEnum values"""
        valid_values = CacheCategoryEnum.values()
        for category in v:
            if category not in valid_values:
                raise ValueError(
                    f"Invalid cache category: '{category}'. "
                    f"Valid values: {valid_values}"
                )
        return v


class CacheClearResponse(BaseSchema):
    """缓存清理结果响应"""

    cleared_categories: list[str] = Field(
        default_factory=list, description="Categories that were cleared"
    )
    cleared_keys: int = Field(0, description="Total number of keys/files cleared")
    cleared_size_bytes: int = Field(0, description="Total size cleared in bytes")
    cleared_size_human: str = Field("0 B", description="Human-readable cleared size")
    duration_ms: int = Field(0, description="Operation duration in milliseconds")


__all__ = [
    "CacheCategorySummary",
    "CacheSummaryResponse",
    "CacheClearRequest",
    "CacheClearResponse",
]
